From d7ae64bff952736c5d8e9a0a5bccdb8b7d07b831 Mon Sep 17 00:00:00 2001
From: Abhishek <apatgaonkar@habana.ai>
Date: Sun, 04 Aug 2024 11:19:38 +0300
Subject: [PATCH] [FSW-2704] fs1: redefine address ranges for io die0

This patch redefines all macros used for io die 0
address ranges. This will allow to validate loadaddr
and dma destination addr in image descriptor.
As scratchpad sram read/write is not happenning
in bootrom, this functionality is removed from
mem_io.h.

Change-Id: Ifa802dca2ffdce7ba7bc3b3fb0b73f3691153e6c
---

diff --git a/arch/fs1/arc/common.c b/arch/fs1/arc/common.c
index 986e0fb..c63c04e 100644
--- a/arch/fs1/arc/common.c
+++ b/arch/fs1/arc/common.c
@@ -164,7 +164,7 @@
 	if (cold_rst_data.arc1_iccm_init_done)
 		return;
 
-	ret = boot_rom_mem_init(HL_ARC1_ICCM_FULL_BASE, ARC1_ICCM_SIZE);
+	ret = boot_rom_mem_init(MNG_ARC_ICCM_FULL_BASE, MNG_ARC_ICCM_SIZE);
 	if (ret) {
 		boot_rom_status_update(PROGRESS_INIT_ARC1_ICCM_INIT_FAILED);
 		return;
diff --git a/arch/fs1/arc/dev_boot_info.h b/arch/fs1/arc/dev_boot_info.h
index dbcae2e..d056125 100644
--- a/arch/fs1/arc/dev_boot_info.h
+++ b/arch/fs1/arc/dev_boot_info.h
@@ -10,7 +10,7 @@
 #include <boot_rom.h>
 
 #define OFFSET_32KB		(32 * 1024)
-#define CONFIG_BOOTINFO_ADDR	(HL_FW_MEM_BASE + HL_SCRATCHPAD_SRAM_OFFSET)
+#define CONFIG_BOOTINFO_ADDR	(HL_FW_MEM_BASE + SCRATCHPAD_SRAM_OFFSET)
 #define CONFIG_BOOTINFO_SIZE	OFFSET_32KB
 
 #endif	/* _DEV_BOOT_INFO_H_ */
diff --git a/arch/fs1/arc/dev_dma.c b/arch/fs1/arc/dev_dma.c
index 83b0ddf..830e5cc 100644
--- a/arch/fs1/arc/dev_dma.c
+++ b/arch/fs1/arc/dev_dma.c
@@ -18,6 +18,16 @@
 #define DMA_SRC_BLOCK_SIZE	1024
 #define DMA_DST_BLOCK_SIZE	4
 
+enum dma_dest_type boot_rom_get_dma_dest_type(uint64_t addr)
+{
+	/* TODO: Add checks for all offsets in FW MEM  for other dies */
+	if (addr >= PSOC_FW_SRAM_FULL_BASE &&
+	    addr < (PSOC_FW_SRAM_FULL_BASE + PSOC_FWSRAM_SIZE))
+		return DMA_DEST_SRAM;
+
+	return DMA_DEST_HBW;
+}
+
 void dma_copy_image_internal(uint16_t image_size, uint32_t flash_offset,
 		uint64_t dest_offset, uint8_t fw_sram_dst, __unused uint8_t restart_crc,
 		__unused uint8_t mask_crc)
diff --git a/arch/fs1/arc/soc.h b/arch/fs1/arc/soc.h
index 1037c0c..f79ba28 100644
--- a/arch/fs1/arc/soc.h
+++ b/arch/fs1/arc/soc.h
@@ -64,51 +64,65 @@
 #define HL_FLASH_OFFSET			0x4000000
 #define HL_FLASH_SIZE			(16 * SIZE_MB)
 
-#define HL_ARC0_ICCM_OFFSET		0x0000000
-#define HL_ARC0_DCCM_OFFSET		0x0010000
-#define HL_ARC1_ICCM_OFFSET		0x0040000
-#define HL_ARC1_DCCM_OFFSET		0x0080000
-#define HL_ARC2_ICCM_OFFSET		0x00C0000
-#define HL_ARC2_DCCM_OFFSET		0x00E0000
-#define HL_ICCM_OFFSET			HL_ARC1_ICCM_OFFSET
-#define HL_SBROM_OFFSET			0x0180000
-#define HL_SCRATCHPAD_SRAM_OFFSET	0x0200000
-#define HL_FWSRAM_OFFSET		0x01A0000
-#define HL_PCIE_FWSRAM_OFFSET		0x0380000
-#define SP_SRAM_BASE			(HL_FW_MEM_BASE + HL_SCRATCHPAD_SRAM_OFFSET)
+/* Below definitions are applicable to IO Die 0 only */
+#define RST_ARC_ICCM_OFFSET		0x0000000
+#define RST_ARC_ICCM_SIZE		SZ_64K
+#define RST_ARC_DCCM_OFFSET		0x0010000
+#define RST_ARC_DCCM_SIZE		SZ_32K
+#define MNG_ARC_ICCM_OFFSET		0x0040000
+#define MNG_ARC_ICCM_SIZE		SZ_256K
+#define MNG_ARC_DCCM_OFFSET		0x0080000
+#define MNG_ARC_DCCM_SIZE		SZ_128K
+#define PID_ARC_ICCM_OFFSET		0x00C0000
+#define PID_ARC_ICCM_SIZE		SZ_128K
+#define PID_ARC_DCCM_OFFSET		0x00E0000
+#define PID_ARC_DCCM_SIZE		SZ_64K
+#define MISC_ARC_ICCM_OFFSET		0x0100000
+#define MISC_ARC_ICCM_SIZE		SZ_256K
+#define MISC_ARC_DCCM_OFFSET		0x0140000
+#define MISC_ARC_DCCM_SIZE		SZ_64K
+#define SBROM_OFFSET			0x0180000
+#define SBROM_SIZE			SZ_128K
+#define PSOC_FWSRAM_OFFSET		0x01A0000
+#define PSOC_FWSRAM_SIZE		SZ_128K
+#define SCRATCHPAD_SRAM_OFFSET		0x0200000
+#define SCRATCHPAD_SRAM_SIZE		SZ_1MB
+#define CXL_ARC_ICCM_OFFSET		0x0300000
+#define CXL_ARC_ICCM_SIZE		SZ_256K
+#define CXL_ARC_DCCM_OFFSET		0x0340000
+#define CXL_ARC_DCCM_SIZE		SZ_256K
+#define PCIE_FWSRAM_OFFSET		0x0380000
+#define PCIE_FWSRAM_SIZE		SZ_64K
 
-#define HL_FW_MEM_BASE_FULL		0x300007FF8800000ull
-#define HL_SP_SRAM_BASE			(HL_FW_MEM_BASE_FULL + HL_SCRATCHPAD_SRAM_OFFSET)
-#define HL_SP_SRAM_SIZE			0x100000
+#define FW_MEM_BASE_FULL		0x300007FF8800000ull
+#define SCRATCHPAD_SRAM_BASE		(HL_FW_MEM_BASE + SCRATCHPAD_SRAM_OFFSET)
+#define SCRATCHPAD_SRAM_BASE_FULL	(FW_MEM_BASE_FULL + SCRATCHPAD_SRAM_OFFSET)
+#define RST_ARC_ICCM_FULL_BASE		(FW_MEM_BASE_FULL + RST_ARC_ICCM_OFFSET)
+#define MNG_ARC_ICCM_FULL_BASE		(FW_MEM_BASE_FULL + MNG_ARC_ICCM_OFFSET)
+#define PID_ARC_ICCM_FULL_BASE		(FW_MEM_BASE_FULL + PID_ARC_ICCM_OFFSET)
+#define MISC_ARC_ICCM_FULL_BASE		(FW_MEM_BASE_FULL + MISC_ARC_ICCM_OFFSET)
+#define CXL_ARC_ICCM_FULL_BASE		(FW_MEM_BASE_FULL + CXL_ARC_ICCM_OFFSET)
+#define PSOC_FW_SRAM_FULL_BASE		(FW_MEM_BASE_FULL + PSOC_FWSRAM_OFFSET)
 
-#define HL_ARC0_ICCM_FULL_BASE		(HL_FW_MEM_BASE_FULL + HL_ARC0_ICCM_OFFSET)
-#define ARC0_ICCM_SIZE			SZ_64K
-#define ARC0_DCCM_SIZE			SZ_32K
-#define HL_ARC1_ICCM_FULL_BASE		(HL_FW_MEM_BASE_FULL + HL_ARC1_ICCM_OFFSET)
-#define ARC1_ICCM_SIZE			SZ_256K
-#define ARC1_DCCM_SIZE			SZ_128K
-#define HL_ARC2_ICCM_FULL_BASE		(HL_FW_MEM_BASE_FULL + HL_ARC2_ICCM_OFFSET)
-#define ARC2_ICCM_SIZE			SZ_128K
-#define ARC2_DCCM_SIZE			SZ_64K
-#define ARC0_MEM_SIZE			(ARC0_ICCM_SIZE + ARC0_DCCM_SIZE)
-#define ARC1_MEM_SIZE			(ARC1_ICCM_SIZE + ARC1_DCCM_SIZE)
-#define ARC2_MEM_SIZE			(ARC2_ICCM_SIZE + ARC2_DCCM_SIZE)
-#define HL_FW_SRAM_FULL_BASE		(HL_FW_MEM_BASE_FULL + HL_FWSRAM_OFFSET)
-#define HL_FW_SRAM_SIZE			SZ_128K
+#define RST_ARC_MEM_SIZE		(RST_ARC_ICCM_SIZE + RST_ARC_DCCM_SIZE)
+#define MNG_ARC_MEM_SIZE		(MNG_ARC_ICCM_SIZE + MNG_ARC_DCCM_SIZE)
+#define PID_ARC_MEM_SIZE		(PID_ARC_ICCM_SIZE + PID_ARC_DCCM_SIZE)
+#define MISC_ARC_MEM_SIZE		(MISC_ARC_ICCM_SIZE + MISC_ARC_DCCM_SIZE)
+#define CXL_ARC_MEM_SIZE		(CXL_ARC_ICCM_SIZE + CXL_ARC_DCCM_SIZE)
 
 #define SPI_MANIFEST_IMG_0_FLASH_OFFSET		(0x0)
-/* manifest 0 boot pointers are DMA trans to ARC1 ICCM at offset 0 */
+/* manifest 0 boot pointers are DMA trans to MNG ARC ICCM at offset 0 */
 #define SPI_MANIFEST_IMG_0_ICCM_OFFSET		(0)
-#define SPI_MANIFEST_IMG_0_OFFSET		(HL_ICCM_OFFSET + SPI_MANIFEST_IMG_0_ICCM_OFFSET)
-#define SPI_MANIFEST_IMG_0_DMA_DEST		(HL_FW_MEM_BASE_FULL + SPI_MANIFEST_IMG_0_OFFSET)
+#define SPI_MANIFEST_IMG_0_OFFSET		(MNG_ARC_ICCM_OFFSET + SPI_MANIFEST_IMG_0_ICCM_OFFSET)
+#define SPI_MANIFEST_IMG_0_DMA_DEST		(FW_MEM_BASE_FULL + SPI_MANIFEST_IMG_0_OFFSET)
 /* 32Bit arc viewpoint address of manifest 0 stored in ICCM */
 #define SPI_MANIFEST_IMG_0_DMA_DEST_ARC		(HL_FW_MEM_BASE + SPI_MANIFEST_IMG_0_OFFSET)
 
 #define SPI_MANIFEST_IMG_1_FLASH_OFFSET		(0x200000)
-/* manifest 1 boot pointers are DMA trans to ARC1 ICCM at offset 64K */
-#define SPI_MANIFEST_IMG_1_ICCM_OFFSET		(64 * 1024)
-#define SPI_MANIFEST_IMG_1_OFFSET		(HL_ICCM_OFFSET + SPI_MANIFEST_IMG_1_ICCM_OFFSET)
-#define SPI_MANIFEST_IMG_1_DMA_DEST		(HL_FW_MEM_BASE_FULL + SPI_MANIFEST_IMG_1_OFFSET)
+/* manifest 1 boot pointers are DMA trans to MNG ARC ICCM at offset 64K */
+#define SPI_MANIFEST_IMG_1_ICCM_OFFSET		SZ_64K
+#define SPI_MANIFEST_IMG_1_OFFSET		(MNG_ARC_ICCM_OFFSET + SPI_MANIFEST_IMG_1_ICCM_OFFSET)
+#define SPI_MANIFEST_IMG_1_DMA_DEST		(FW_MEM_BASE_FULL + SPI_MANIFEST_IMG_1_OFFSET)
 /* 32Bit arc viewpoint address of manifest 1 stored in ICCM */
 #define SPI_MANIFEST_IMG_1_DMA_DEST_ARC		(HL_FW_MEM_BASE + SPI_MANIFEST_IMG_1_OFFSET)
 
diff --git a/common/dma.c b/common/dma.c
index 9fac28f..37e152e 100644
--- a/common/dma.c
+++ b/common/dma.c
@@ -11,17 +11,6 @@
 #include <status.h>
 #include "dma.h"
 
-enum dma_dest_type boot_rom_get_dma_dest_type(uint64_t addr)
-{
-	/* TODO: Add checks for all offsets in FW MEM */
-	if (addr >= HL_FW_SRAM_FULL_BASE &&
-	    addr < (HL_FW_SRAM_FULL_BASE + HL_FW_SRAM_SIZE))
-		return DMA_DEST_SRAM;
-
-	/* */
-	return DMA_DEST_HBW;
-}
-
 int boot_rom_process_dma_err(int err, enum boot_rom_progress dma_sts_base)
 {
 	if (err != DMA_SUCCESS) {
diff --git a/include/mem_io.h b/include/mem_io.h
index de1ba6f..763ca3a 100644
--- a/include/mem_io.h
+++ b/include/mem_io.h
@@ -13,7 +13,6 @@
 #define HL_FLASH_BASE				(HL_GEN_MEM_BASE + HL_FLASH_OFFSET)
 #define HL_SBROM_BASE				(HL_FW_MEM_BASE + HL_SBROM_OFFSET)
 #define HL_FWSRAM_BASE				(HL_FW_MEM_BASE + HL_FWSRAM_OFFSET)
-#define HL_SCRATCHPAD_SRAM_BASE			(HL_FW_MEM_BASE + HL_SCRATCHPAD_SRAM_OFFSET)
 
 /* In ARC address manipulation BIT[22:24] are for die id */
 #define HL_ARC_ADDR_DIE_ID_SHIFT		25
@@ -46,18 +45,6 @@
 static inline void regmap_writel(uint32_t val, uint32_t offset)
 {
 	base_writel(val, (ARC_REG_UPD(offset, HL_CFG_BASE)));
-}
-
-static inline uint32_t sp_sram_readl(uint32_t offset)
-{
-	uint32_t reg = base_readl(HL_SCRATCHPAD_SRAM_BASE + offset);
-
-	return reg;
-}
-
-static inline void sp_sram_writel(uint32_t val, uint32_t offset)
-{
-	base_writel(val, HL_SCRATCHPAD_SRAM_BASE + offset);
 }
 
 static inline uint8_t regmap_readb(uint32_t offset)
